File: C# Fundamentals/C# OOP Advanced/09.ReflectionN_Attributes/Ex/P03_BarraksWars/Core/Commands/AddCommand.cs
namespace _03BarracksFactory.Core.Commands
{
    using _03BarracksFactory.Attributes;
    using _03BarracksFactory.Contracts;

    public class AddCommand : Command
    {
        [Inject]
        private IRepository repository;

        [Inject]
        private IUnitFactory unitFactory;

        public AddCommand(string[] data) : base(data)
        {
        }

        public override string Execute()
        {
            string unitType = Data[1];
            IUnit unitToAdd = this.unitFactory.CreateUnit(unitType);
            this.repository.AddUnit(unitToAdd);
            string output = unitType + " added!";
            return output;
        }
    }
}

File: Java/01.Advanced/02.OOP/15.ReflectionAnnotation/Ex/src/barracksWars/core/commands/AddCommand.java
package barracksWars.core.commands;

import barracksWars.annotations.Inject;
import barracksWars.interfaces.Repository;
import barracksWars.interfaces.Unit;
import barracksWars.interfaces.UnitFactory;
import jdk.jshell.spi.ExecutionControl;

public class AddCommand extends Command{

    @Inject
    private UnitFactory unitFactory;

    @Inject
    private Repository repository;

    public AddCommand(String[] data) {
        super(data);
    }

    @Override
    public String execute() throws ExecutionControl.NotImplementedException {
        String unitType = getData()[1];
        Unit unitToAdd = this.unitFactory.createUnit(unitType);
        this.repository.addUnit(unitToAdd);
        return unitType + " added!";
    }
}
